avocado/core/suite.py

```diff
@@ -98,7 +98,8 @@ def __init__(self, name, config=None, tests=None, job_config=None,
             self.config.update(config)
 
         # Update the config of runnables.
-        if config.get('run.test_runner') == 'nrunner' and self.tests:
+        test_runner = self.config.get('run.test_runner') or 'nrunner'
```

Comment:
We have a [default value](https://github.com/avocado-framework/avocado/blob/master/avocado/plugins/run.py#L106) for the `run.test_runner` inside config, so I think we should reused here instead of setting new one. This is related to all the changes in this commit.

Reply:
> We have a [default value](https://github.com/avocado-framework/avocado/blob/master/avocado/plugins/run.py#L106) for the `run.test_runner` inside config, so I think we should reused here instead of setting new one. This is related to all the changes in this commit.

You're right.  I was on the fence about always registering `run.test_runner` (say in `avocado/core/__init__.py`, but I guess that's the way to go.